jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java

```diff
@@ -160,7 +159,22 @@ static boolean isBrokenPipe(IOException original) {
    * @throws RegistryException for known exceptions when interacting with the registry
    */
   T call() throws IOException, RegistryException {
-    return callWithAllowInsecureRegistryHandling(initialRequestUrl);
+    try {
+      return callWithAllowInsecureRegistryHandling(initialRequestUrl);
+
+    } catch (IOException ex) {
+      String registry = registryEndpointRequestProperties.getServerUrl();
+      String repository = registryEndpointRequestProperties.getImageName();
+      logError("I/O error for image [" + registry + "/" + repository + "]:");
+      logError("    " + ex.getMessage());
+      if (isBrokenPipe(ex)) {
+        logError(
```

Comment:
What's the reasoning behind logging the message separately from the exception instead of throwing an exception with a custom message? Just to log the error message even if the exception is caught elsewhere?

Reply:
Not just to log. The main reason is to tell the user for which image (registry/repository) the error was for. That's the original issue in #1895, where the user had to spend a lot of time figuring out that it was the connection to `gcr.io` that was timing out. The user didn't know anything about what the default base image is and where it is hosted.

Now, we can get all sorts of `IOException`s (extended by `SocketTimeoutException`, `SSLException`, `ConnectException`, `NoHttpResponseException`, ..), and we can't mutate their fixed message to include the image information. Creating another `IOException` with our custom message and wrapping the cause just for the sake of customizing the message doesn't like a good idea, as it hides the cause such that, e.g., a `catch (SSLException ex)` block no longer works at a higher stack.

Reply:
Oh I see now.